test/Microsoft.Identity.Abstractions.Tests/AquireTokenOptionsTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Xunit;
+
+namespace Microsoft.Identity.Abstractions.Tests
+{
+    public class AquireTokenOptionsTests
+    {
+        [Fact]
+        public void ManagedIdentitySystemAssigned()
+        {
+            // App token from a system-assigned managed identity.
+            // -------------------------------------------------
+            // https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview
+            /*
+            // <managedidentity_json>
+            {
+                "AquireTokenOptions": {
+                    "ManagedIdentity": {
+                        "Source": "SystemAssigned"
+                    }
+                }
+            }
+            // </managedidentitysystem_json>
+            */
+
+            // <managedidentity_csharp>
+            ManagedIdentityDescription managedIdentityDescription = new ManagedIdentityDescription
+            {
+                Source = ManagedIdentitySource.SystemAssigned
+            };
+
+            AcquireTokenOptions acquireTokenOptions = new AcquireTokenOptions
+            {
+                ManagedIdentity = managedIdentityDescription
+            };
+            // </managedidentitysystem_csharp>
+
+            Assert.Equal(ManagedIdentitySource.SystemAssigned, acquireTokenOptions.ManagedIdentity.Source);
+            Assert.Null(acquireTokenOptions.ManagedIdentity.UniqueIdType);
+            Assert.Null(acquireTokenOptions.ManagedIdentity.UniqueIdValue);
+        }
+
+        [Fact]
+        public void ManagedIdentityUserAssigned()
+        {
+            // App token from a user-assigned managed identity.
+            // -------------------------------------------------
+            // https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview
+            /*
+            // <managedidentityuser_json>
+            {
+                "AquireTokenOptions": {
```

Comment:
With the proposal above we'd have:

```json
              "AquireTokenOptions": {
                    "ManagedIdentity": {
                        "IdType": "UserAssignedManagedIdentityClientId"
                        "Id": "[InsertRelevantValueForTheManagedIdentityResource]"
                    }
                }
```

Reply:
I was worried about scalability in the future if new types of managed identities are added, new apis with different methods of access or different guids. An enum should only define one property ideally and this would define two both user assigned and the guid to use.

Reply:
there won't be new types of managed identities :-). Either it's provided by the infrastructure, or it's provided by the developer (user). This is a partition